packages/react-native/src/handlers/UnhandledExceptionHandler.ts

```diff
@@ -0,0 +1,113 @@
+import { BacktraceReport } from '@backtrace-labs/sdk-core';
+import { BacktraceClient } from '../BacktraceClient';
+import { hermes } from '../common/hermesHelper';
+import { type ExceptionHandler } from './ExceptionHandler';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const rejectionTracking = require('promise/setimmediate/rejection-tracking');
+
+export class UnhandledExceptionHandler implements ExceptionHandler {
+    protected enabled = true;
+    public captureManagedErrors(client: BacktraceClient) {
+        const globalErrorHandler = ErrorUtils.getGlobalHandler();
+        ErrorUtils.setGlobalHandler((error: Error, fatal?: boolean) => {
+            if (!this.enabled) {
+                return;
+            }
+            client.send(error, {
+                'error.type': 'Unhandled exception',
+                fatal,
+            });
+            globalErrorHandler(error, fatal);
+        });
+    }
+
+    public captureUnhandledPromiseRejections(client: BacktraceClient) {
+        const hermesInternal = hermes();
+
+        if (hermesInternal?.hasPromise?.() && hermesInternal?.enablePromiseRejectionTracker) {
+            hermesInternal.enablePromiseRejectionTracker({
+                allRejections: true,
+                onUnhandled: (id: number, rejection: Error | object = {}) => {
+                    if (!this.enabled) {
+                        return;
+                    }
+                    client.send(
+                        new BacktraceReport(
+                            rejection as Error,
+                            {
+                                'error.type': 'Unhandled exception',
+                                unhandledPromiseRejectionId: id,
+                            },
+                            [],
+                            {
+                                classifiers: ['UnhandledPromiseRejection'],
+                                skipFrames: rejection instanceof Error ? 0 : 1,
```

Comment:
On line 37 you cast to `Error`, but you check here whether `rejection` is `instanceof Error`. So it could be something else than error? Maybe we should employ additional checks on line 37?

Also applies to `rejectionTracking` mechanism.

Reply:
Adjusted the code. Thank you!